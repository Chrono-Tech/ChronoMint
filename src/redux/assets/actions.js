import contractsManagerDAO from 'dao/ContractsManagerDAO'

export const ASSETS_LIST = 'assets/LIST'

export const listAssets = () => async (dispatch) => {
  const dao = await contractsManagerDAO.getAssetsManagerDAO()
  dispatch({type: ASSETS_LIST, list: await dao.getList()})
}
