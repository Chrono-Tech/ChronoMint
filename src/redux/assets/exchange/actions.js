import type ExchangeManagerDAO from 'dao/ExchangeManagerDAO'
import type ExchangeModel from 'models/ExchangeModel'
import type TokenModel from 'models/TokenModel'
import contractsManagerDAO from 'dao/ContractsManagerDAO'

export const EXCHANGE_MANAGER_LIST = 'exchangeManager/LIST'
export const EXCHANGE_MANAGER_SET = 'exchangeManager/SET'
export const EXCHANGE_MANAGER_REMOVE = 'exchangeManager/REMOVE'

const getDAO = async (): Promise<ExchangeManagerDAO> => await contractsManagerDAO.getExchangeManagerDAO()

export const listExchanges = () => async (dispatch) => {
  const dao = await getDAO()
  dispatch({type: EXCHANGE_MANAGER_LIST, list: await dao.getList()})
}

export const addExchange = (address) => async () => {
  const exchange = new ExchangeModel({address, isFetching: true})
  dispatch({type: EXCHANGE_MANAGER_SET, exchange})
  try {
    const dao = await getDAO()
    await dao.addExchange(address)
  } catch (e) {
    // no rollback
  }
}

export const createExchange = (token: TokenModel) => async () => {
  try {
    const dao = await getDAO()
    await dao.createExchange(token)
  } catch (e) {
    // no rollback
  }
}

export const editExchange = (exchange: ExchangeModel, newExchange: ExchangeModel) => async () => {
  try {
    const dao = await getDAO()
    await dao.editExchange(exchange, newExchange)
  } catch (e) {
    // no rollback
  }
}

export const removeExchange = (exchange: ExchangeModel) => async () => {
  try {
    const dao = await getDAO()
    await dao.removeExchange(exchange)
  } catch (e) {
    // no rollback
  }
}

// TODO @bshevchenko: MINT-230 AssetsManager
// export const addOwner = (exchange: ExchangeModel, account) => async () => {
//   try {
//     const dao = await getDAO()
//     await dao.addOwner(exchange, account)
//   } catch (e) {
//     // no rollback
//   }
// }
//
// export const removeOwner = (exchange: ExchangeModel, account) => async () => {
//   try {
//     const dao = await getDAO()
//     await dao.removeOwner(exchange, account)
//   } catch (e) {
//     // no rollback
//   }
// }
