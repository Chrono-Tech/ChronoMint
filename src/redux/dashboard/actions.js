import LHTProxyDAO from '../../dao/LHTProxyDAO'
import UserStorageDAO from '../../dao/UserStorageDAO'

import {
  DASHBOARD_TOTAL_LHT_FETCH,
  DASHBOARD_TOTAL_LHT,
  DASHBOARD_TOTAL_MEMBERS_FETCH,
  DASHBOARD_TOTAL_MEMBERS
} from './reducer'

export const updateTotalLHT = () => (dispatch) => {
  dispatch({type: DASHBOARD_TOTAL_LHT_FETCH})
  return LHTProxyDAO.totalSupply()
    .then(balance => {
      dispatch({type: DASHBOARD_TOTAL_LHT, payload: balance})
    })
}

export const updateTotalMembers = () => (dispatch) => {
  dispatch({type: DASHBOARD_TOTAL_MEMBERS_FETCH})
  return UserStorageDAO.usersTotal()
    .then(number => {
      dispatch({type: DASHBOARD_TOTAL_MEMBERS, payload: number})
    })
}
