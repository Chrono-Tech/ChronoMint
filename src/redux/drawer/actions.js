/**
 * Copyright 2017â€“2018, LaborX PTY
 * Licensed under the AGPL Version 3 license.
 */

import {
  DRAWER_HIDE,
  DRAWER_TOGGLE,
} from './constants'

export const drawerToggle = () => (dispatch) => {
  dispatch({ type: DRAWER_TOGGLE })
}
export const drawerHide = () => (dispatch) => {
  dispatch({ type: DRAWER_HIDE })
}
