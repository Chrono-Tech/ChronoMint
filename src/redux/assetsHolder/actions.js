import { DUCK_SESSION } from '@/redux/session/actions'
import contractsManagerDAO from 'dao/ContractsManagerDAO'
import TIMEHolderDAO from 'dao/TIMEHolderDAO'
import Amount from 'models/Amount'
import ApprovalNoticeModel from 'models/notices/ApprovalNoticeModel'
import TokenModel from 'models/tokens/TokenModel'
import { DUCK_TOKENS } from 'redux/tokens/actions'
import tokenService, { EVENT_NEW_TOKEN } from 'services/TokenService'

export const DUCK_ASSETS_HOLDER = 'assetsHolder'

export const TIME_HOLDER_INIT = 'timeHolder/INIT'
export const TIME_HOLDER_TIME_ADDRESS = 'timeHolder/timeAddress'
export const TIME_HOLDER_WALLET_ADDRESS = 'timeHolder/timeHolderWalletAddress'
export const TIME_HOLDER_ADDRESS = 'timeHolder/timeHolderAddress'

let timeHolderDAO: TIMEHolderDAO = null

const subscribeOnTokens = (token: TokenModel) => async (dispatch, getState) => {
  if (token.symbol() !== 'TIME') {
    return
  }

  dispatch({ type: TIME_HOLDER_TIME_ADDRESS, address: token.address() })

  // init
  const [ timeHolderAddress, timeHolderWalletAddress ] = await Promise.all([
    timeHolderDAO.getAddress(),
    timeHolderDAO.getWalletAddress(),
  ])

  dispatch({ type: TIME_HOLDER_ADDRESS, address: timeHolderAddress })
  dispatch({ type: TIME_HOLDER_WALLET_ADDRESS, address: timeHolderWalletAddress })

  // TODO @dkchv: review again
  const contractNames = {}
  contractNames[ timeHolderAddress ] = 'TIME Holder'
  ApprovalNoticeModel.setContractNames(contractNames)

  // update allowance
  const timeDAO = tokenService.getDAO(token.id())
  const { account } = getState().get(DUCK_SESSION)
  const [ timeHolderAllowance, timeHolderWalletAllowance ] = await Promise.all([
    timeDAO.getAccountAllowance(timeHolderAddress, account),
    timeDAO.getAccountAllowance(timeHolderWalletAddress, account),
  ])

  console.log('--assetHolder: ', +timeHolderAllowance, +timeHolderWalletAllowance)

  // token = token
  //   .setAllowance(timeHolderAddress, timeHolderAllowance)
  //   .setAllowance(timeHolderWalletAddress, timeHolderWalletAllowance)
}

export const initAssetsHolder = () => async (dispatch, getState) => {
  if (getState().get(DUCK_ASSETS_HOLDER).isInited()) {
    return
  }
  dispatch({ type: TIME_HOLDER_INIT, inInited: true })

  const { account } = getState().get(DUCK_SESSION)
  timeHolderDAO = await contractsManagerDAO.getTIMEHolderDAO()
  const [ assets, deposit ] = await Promise.all([
    timeHolderDAO.getSharesContract(),
    timeHolderDAO.getDeposit(account),
  ])
  console.log('--deposit', assets, +deposit)

  // subscribe to tokens
  const callback = (token) => dispatch(subscribeOnTokens(token))
  tokenService.on(EVENT_NEW_TOKEN, callback)
  // fetch for existing tokens
  const tokens = getState().get(DUCK_TOKENS)
  tokens.list().forEach(callback)
}

export const depositAsset = (amount: Amount, token: TokenModel) => async (dispatch) => {
  // const amountBN = new BigNumber(amount)

  // dispatch(balanceMinus(amountBN, token))
  const assetDAO = tokenService.getDAO(token)

  try {
    await timeHolderDAO.deposit(amount, assetDAO)
  } catch (e) {
    // eslint-disable-next-line
    console.error('deposit error', e.message)
  } finally {
    // compensation for update in watchTransfer
    // dispatch(balancePlus(amount, token))
  }
}

export const withdrawAsset = (amount: Amount, token: TokenModel) => async () => {
  // const amountBN = new BigNumber(amount)

  // dispatch(depositMinus(amountBN))
  const assetDAO = tokenService.getDAO(token)

  try {
    await timeHolderDAO.withdraw(amount, assetDAO)
  } catch (e) {
    // eslint-disable-next-line
    console.error('withdraw error', e.message)
  } finally {
    // dispatch(depositPlus(amountBN))
  }
}
