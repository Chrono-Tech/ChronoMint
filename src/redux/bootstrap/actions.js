import ls from 'utils/LocalStorage'
import { LOCAL_ID } from 'network/settings'

import {
  checkMetaMask, checkLocalSession, restoreLocalSession, createNetworkSession,
  checkTestRPC
} from '../network/actions'
import { login } from '../session/actions'

export const bootstrap = (relogin = true) => async (dispatch) => {
  dispatch(checkMetaMask())
  dispatch(checkTestRPC())

  if (!relogin) {
    return
  }

  const localAccount = ls.getLocalAccount()
  const isPassed = await dispatch(checkLocalSession(localAccount))
  if (isPassed) {
    await dispatch(restoreLocalSession(localAccount))
    dispatch(createNetworkSession(localAccount, LOCAL_ID, LOCAL_ID))
    dispatch(login(localAccount))
  } else {
    // eslint-disable-next-line
    console.warn('Can\'t restore local session')
  }
}
