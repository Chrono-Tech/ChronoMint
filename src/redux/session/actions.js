import { push, replace } from 'react-router-redux'
import contractsManagerDAO from 'dao/ContractsManagerDAO'
import networkService from '@chronobank/login/network/NetworkService'
import ProfileModel from 'models/ProfileModel'
import { bootstrap } from 'redux/bootstrap/actions'
import { cbeWatcher, watcher } from 'redux/watcher/actions'
import { initWallet } from 'redux/wallet/actions'
import { removeWatchersUserMonitor } from 'redux/userMonitor/actions'
import { watchStopMarket } from 'redux/market/action'
import ls from 'utils/LocalStorage'

export const DUCK_SESSION = 'session'

export const SESSION_CREATE = 'session/CREATE'
export const SESSION_DESTROY = 'session/DESTROY'

export const SESSION_PROFILE = 'session/PROFILE'
export const SESSION_PROFILE_UPDATE = 'session/PROFILE_UPDATE'

export const DEFAULT_USER_URL = '/dashboard'
export const DEFAULT_CBE_URL = '/dashboard'

export const createSession = ({ account, provider, network, dispatch }) => {
  ls.createSession(account, provider, network)
  dispatch({ type: SESSION_CREATE, account })
}

export const destroySession = ({ lastURL, dispatch }) => {
  ls.setLastURL(lastURL)
  ls.destroySession()
  dispatch({ type: SESSION_DESTROY })
}

export const logout = () => async (dispatch) => {
  try {
    dispatch(removeWatchersUserMonitor())
    await dispatch(watchStopMarket())
    await networkService.destroyNetworkSession(`${window.location.pathname}${window.location.search}`)
    await dispatch(push('/'))
    await dispatch(bootstrap(false))
  } catch (e) {
    // eslint-disable-next-line
    console.warn('logout error:', e)
  }
}

export const login = (account) => async (dispatch, getState) => {
  if (!getState().get(DUCK_SESSION).isSession) {
    // setup and check network first and create session
    throw new Error('Session has not been created')
  }

  const dao = await contractsManagerDAO.getUserManagerDAO()
  const [ isCBE, profile, memberId ] = await Promise.all([
    dao.isCBE(account),
    dao.getMemberProfile(account),
    dao.getMemberId(account),
  ])

  // TODO @bshevchenko: PendingManagerDAO should receive member id from redux state
  const pmDAO = await contractsManagerDAO.getPendingManagerDAO()
  pmDAO.setMemberId(memberId)

  dispatch({ type: SESSION_PROFILE, profile, isCBE })

  const defaultURL = isCBE ? DEFAULT_CBE_URL : DEFAULT_USER_URL
  dispatch(initWallet())
  dispatch(watcher())
  isCBE && dispatch(cbeWatcher())
  dispatch(replace((isCBE && ls.getLastURL()) || defaultURL))
}

export const updateUserProfile = (newProfile: ProfileModel) => async (dispatch, getState) => {
  const { isSession, account, profile } = getState().get(DUCK_SESSION)
  if (!isSession) {
    // setup and check network first and create session
    throw new Error('Session has not been created')
  }
  dispatch({ type: SESSION_PROFILE_UPDATE, profile: newProfile })
  try {
    const dao = await contractsManagerDAO.getUserManagerDAO()
    await dao.setMemberProfile(account, newProfile)
  } catch (e) {
    // eslint-disable-next-line
    console.error('update profile error', e.message)
    dispatch({ type: SESSION_PROFILE_UPDATE, profile })
  }
}
