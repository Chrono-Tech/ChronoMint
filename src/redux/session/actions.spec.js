import Immutable from 'immutable'
import { accounts, mockStore } from 'specsInit'
import ProfileModel from 'models/ProfileModel'
import { LOCAL_ID } from 'network/settings'
import { MARKET_INIT } from 'redux/market/action'
import { WATCHER, WATCHER_CBE } from 'redux/watcher/actions'
import ls from 'utils/LocalStorage'
import * as a from './actions'

let store

const profile = new ProfileModel({ name: 'profile1' })
// TODO let userProfile: ProfileModel

const REPLACE_METHOD = 'replace'
const MOCK_LAST_URL = '/test-last-url'

const routerAction = (route, method = 'push') => ({
  type: '@@router/CALL_HISTORY_METHOD',
  payload: { args: [route], method },
})

const emptySessionMock = new Immutable.Map({
  market: {
    rates: {},
    lastMarket: {},
  },
})

const cbeSessionMock = new Immutable.Map({
  market: {
    rates: {},
    lastMarket: {},
  },
  session: {
    isSession: true,
    account: accounts[0],
  },
})

const userSessionMock = new Immutable.Map({
  market: {
    rates: {},
    lastMarket: {},
  },
  session: {
    isSession: true,
    account: accounts[5],
  },
})

describe('session actions', () => {
  beforeEach(() => {
    // override common cbe session
    ls.destroySession()
  })

  it('should create session', () => {
    store = mockStore(emptySessionMock)
    store.dispatch(a.createSession(accounts[0]))
    expect(store.getActions()).toEqual([
      { type: a.SESSION_CREATE, account: accounts[0] },
    ])
  })

  it('should destroy session', () => {
    store = mockStore(emptySessionMock)
    store.dispatch(a.destroySession())
    expect(store.getActions()).toEqual([
      { type: a.SESSION_DESTROY },
    ])
  })

  it('should not login without session', async () => {
    store = mockStore(emptySessionMock)
    let error = null
    try {
      await store.dispatch(a.login(accounts[0]))
    } catch (e) {
      error = e
    }
    expect(error).not.toBeNull()
  })

  it('should not update profile without session', async () => {
    store = mockStore(emptySessionMock)
    let error = null
    try {
      await store.dispatch(a.updateUserProfile(accounts[0]))
    } catch (e) {
      error = e
    }
    expect(error).not.toBeNull()
  })

  it.skip('should update profile', async () => {
    const store = mockStore(cbeSessionMock)
    ls.createSession(accounts[1], LOCAL_ID, LOCAL_ID)
    await store.dispatch(a.updateUserProfile(profile))

    expect(store.getActions()).toEqual([
      { type: a.SESSION_PROFILE_UPDATE, profile },
    ])
  })

  it('should login CBE and start watcher with cbeWatcher and go to last url', async () => {
    store = mockStore(cbeSessionMock)
    ls.createSession(accounts[0], LOCAL_ID, LOCAL_ID)
    ls.setLastURL(MOCK_LAST_URL)
    store.clearActions()
    await store.dispatch(a.login(accounts[0]))

    const actions = store.getActions()
    // TODO expect(actions).toContainEqual({type: a.SESSION_PROFILE, profile: userProfile, isCBE: true})
    expect(actions).toContainEqual({ type: WATCHER })
    expect(actions).toContainEqual({ type: WATCHER_CBE })
    expect(actions).toContainEqual(routerAction(MOCK_LAST_URL, REPLACE_METHOD))
  })

  it('should login CBE and go to default page (/cbe)', async () => {
    store = mockStore(cbeSessionMock)
    ls.createSession(accounts[0], LOCAL_ID, LOCAL_ID)
    store.clearActions()

    await store.dispatch(a.login(accounts[0]))

    const actions = store.getActions()
    // TODO expect(actions).toContainEqual({type: a.SESSION_PROFILE, profile: userProfile, isCBE: true})
    expect(actions).toContainEqual({ type: WATCHER })
    expect(actions).toContainEqual({ type: WATCHER_CBE })
    expect(actions).toContainEqual(routerAction(a.DEFAULT_CBE_URL, REPLACE_METHOD))
  })

  it('should login USER and go to default url (/walelt)', async () => {
    store = mockStore(userSessionMock)
    ls.createSession(accounts[5], LOCAL_ID, LOCAL_ID)
    store.clearActions()

    await store.dispatch(a.login(accounts[5]))

    const actions = store.getActions()
    expect(actions).toContainEqual({ type: a.SESSION_PROFILE, profile: new ProfileModel(), isCBE: false })
    expect(actions).toContainEqual({ type: WATCHER })
    expect(actions).not.toContainEqual({ type: WATCHER_CBE })
    expect(actions).toContainEqual(routerAction(a.DEFAULT_USER_URL, REPLACE_METHOD))
  })

  it('should logout', async () => {
    store = mockStore(userSessionMock)
    ls.createSession(accounts[5], LOCAL_ID, LOCAL_ID)
    store.clearActions()

    await store.dispatch(a.logout())

    expect(store.getActions()).toEqual([
      { type: MARKET_INIT, isInited: false },
      { type: a.SESSION_DESTROY },
      routerAction('/'),
    ])
  })
})
