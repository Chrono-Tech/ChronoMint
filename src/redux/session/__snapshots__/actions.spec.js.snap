// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`session actions should create session 1`] = `
[
  {
    "account": "0xb7237dd7dacd9da8e489b57d4631fc33342ff75f",
    "type": "session/CREATE"
  }
]
`;

exports[`session actions should destroy session 1`] = `
[
  {
    "type": "session/DESTROY"
  }
]
`;

exports[`session actions should login CBE and go to default page (/cbe) 1`] = `
[
  {
    "isCBE": true,
    "profile": {
      "assetsManager": {},
      "company": null,
      "email": null,
      "icon": null,
      "id": null,
      "name": null,
      "timestamp": null,
      "tokens": [],
      "url": null
    },
    "type": "session/PROFILE"
  },
  {
    "type": "WALLET/switch_wallet",
    "wallet": {
      "address": null,
      "bccAddress": null,
      "btcAddress": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isMainWallet": true,
      "isMultisig": false,
      "isPending": false,
      "isTIMERequired": true,
      "nemAddress": null,
      "timeAddress": null,
      "timeDeposit": "0",
      "timestamp": null,
      "tokens": {},
      "transactionHash": null,
      "transactions": {
        "endOfList": false,
        "id": null,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isPending": false,
        "list": {},
        "selected": null,
        "timestamp": null,
        "transactionHash": null
      }
    }
  },
  {
    "type": "exchange/GET_DATA_START"
  },
  {
    "type": "exchange/EXCHANGE_GET_TOKENS_LIST_START"
  },
  {
    "type": "AssetsManager/GET_ASSETS_MANAGER_COUNTS_START"
  },
  {
    "type": "AssetsManager/GET_TRANSACTIONS_START"
  },
  {
    "isInited": true,
    "type": "market/INIT"
  },
  {
    "type": "mainWallet/TOKENS_FETCH"
  },
  {
    "type": "watcher/USER"
  },
  {
    "type": "watcher/CBE"
  },
  {
    "payload": {
      "args": [
        "/dashboard"
      ],
      "method": "replace"
    },
    "type": "@@router/CALL_HISTORY_METHOD"
  }
]
`;

exports[`session actions should login CBE and start watcher with cbeWatcher and go to last url 1`] = `
[
  {
    "isCBE": true,
    "profile": {
      "assetsManager": {},
      "company": null,
      "email": null,
      "icon": null,
      "id": null,
      "name": null,
      "timestamp": null,
      "tokens": [],
      "url": null
    },
    "type": "session/PROFILE"
  },
  {
    "type": "WALLET/switch_wallet",
    "wallet": {
      "address": null,
      "bccAddress": null,
      "btcAddress": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isMainWallet": true,
      "isMultisig": false,
      "isPending": false,
      "isTIMERequired": true,
      "nemAddress": null,
      "timeAddress": null,
      "timeDeposit": "0",
      "timestamp": null,
      "tokens": {},
      "transactionHash": null,
      "transactions": {
        "endOfList": false,
        "id": null,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isPending": false,
        "list": {},
        "selected": null,
        "timestamp": null,
        "transactionHash": null
      }
    }
  },
  {
    "type": "exchange/GET_DATA_START"
  },
  {
    "type": "exchange/EXCHANGE_GET_TOKENS_LIST_START"
  },
  {
    "type": "AssetsManager/GET_ASSETS_MANAGER_COUNTS_START"
  },
  {
    "type": "AssetsManager/GET_TRANSACTIONS_START"
  },
  {
    "isInited": true,
    "type": "market/INIT"
  },
  {
    "type": "mainWallet/TOKENS_FETCH"
  },
  {
    "type": "watcher/USER"
  },
  {
    "type": "watcher/CBE"
  },
  {
    "payload": {
      "args": [
        "/test-last-url"
      ],
      "method": "replace"
    },
    "type": "@@router/CALL_HISTORY_METHOD"
  }
]
`;

exports[`session actions should login USER and go to default url (/wallet) 1`] = `
[
  {
    "isCBE": false,
    "profile": {
      "assetsManager": {},
      "company": null,
      "email": null,
      "icon": null,
      "id": null,
      "name": null,
      "timestamp": null,
      "tokens": [],
      "url": null
    },
    "type": "session/PROFILE"
  },
  {
    "type": "WALLET/switch_wallet",
    "wallet": {
      "address": null,
      "bccAddress": null,
      "btcAddress": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isMainWallet": true,
      "isMultisig": false,
      "isPending": false,
      "isTIMERequired": true,
      "nemAddress": null,
      "timeAddress": null,
      "timeDeposit": "0",
      "timestamp": null,
      "tokens": {},
      "transactionHash": null,
      "transactions": {
        "endOfList": false,
        "id": null,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isPending": false,
        "list": {},
        "selected": null,
        "timestamp": null,
        "transactionHash": null
      }
    }
  },
  {
    "type": "exchange/GET_DATA_START"
  },
  {
    "type": "exchange/EXCHANGE_GET_TOKENS_LIST_START"
  },
  {
    "type": "AssetsManager/GET_ASSETS_MANAGER_COUNTS_START"
  },
  {
    "type": "AssetsManager/GET_TRANSACTIONS_START"
  },
  {
    "isInited": true,
    "type": "market/INIT"
  },
  {
    "type": "mainWallet/TOKENS_FETCH"
  },
  {
    "type": "watcher/USER"
  },
  {
    "payload": {
      "args": [
        "/dashboard"
      ],
      "method": "replace"
    },
    "type": "@@router/CALL_HISTORY_METHOD"
  }
]
`;

exports[`session actions should logout 1`] = `
[
  {
    "isInited": false,
    "type": "market/INIT"
  },
  {
    "payload": {
      "args": [
        "/"
      ],
      "method": "push"
    },
    "type": "@@router/CALL_HISTORY_METHOD"
  }
]
`;
